src/cargo/core/compiler/timings.js

```diff
@@ -23,6 +23,36 @@ let UNIT_COORDS = {};
 let REVERSE_UNIT_DEPS = {};
 let REVERSE_UNIT_RMETA_DEPS = {};
 
+function scale_to_graph_width(scale) {
+    // Cap the size of the graph. It is hard to view if it is too large, and
+    // browsers may not render a large graph because it takes too much memory.
+    // 4096 is still ridiculously large, and probably won't render on mobile
+    // browsers, but should be ok for many desktop environments.
+    return Math.min(200 + scale * 20, 4096);
+}
+
+// Guess a reasonable initial scale value to fit the graphs to the width of the user's window.
+function init_graph_scale() {
+    const width = window.innerWidth;
+    // Leave some margin around the chart, to account for axis labels
+    const max_available_space = Math.max(200, width - 150);
+
+    for (let scale = 1; scale < 100; scale++) {
+        const width = scale_to_graph_width(scale);
+
+        // If we ran out of space, return the previous scale
+        if (width > max_available_space) {
+            return Math.max(1, scale);
```

Comment:
Isn't this `max` redundant?

Reply:
Right, this should have actually been `scale - 1`, then the max makes sense. Thanks!